haskell/haskell.bzl

```diff
@@ -326,6 +332,61 @@ $ bazel-bin/.../hello-lib-repl  # run the script
 ```
 """
 
+def _haskell_proto_library_impl(ctx):
+
+  source_files = depset()
+  for dep in ctx.attr.deps:
+    if HaskellProtobufInfo in dep:
+      source_files = depset(transitive = [
+        source_files,
+        dep[HaskellProtobufInfo].source_files,
+      ])
+
+  return _haskell_library_impl(
+    ctx,
+    use_these_srcs=source_files,
+  )
+
+haskell_proto_library = rule(
+  _haskell_proto_library_impl,
+  attrs = dict(
+    _haskell_common_attrs,
+    hidden_modules = attr.string_list(
```

Comment:
Is `hidden_modules` still used in this PR?

Reply:
I decided that having the same attribute set will allow us to have fewer extra conditionals in `_haskell_library_impl`. If we remove `hidden_modules`, we'll need to guard every lookup of that attribute. On the other hand, if we keep `hidden_modules`, I don't see any problem with that.